nautobot/dcim/filters.py

```diff
@@ -132,7 +132,7 @@ class RegionFilterSet(NautobotFilterSet, NameSlugSearchFilterSet):
         to_field_name="slug",
         label="Parent region (slug)",
     )
-    children = NaturalKeyOrPKMultipleChoiceFilter(
+    children = TreeNodeMultipleChoiceFilter(
```

Comment:
I agree that children should be `NaturalKeyOrPKMultipleChoiceFilter`.

I don't think descendants makes sense for `TreeNodeMultipleChoiceFilter` because that filter is using `get_descendants()`. For it to work properly as a filter we would need to return ancestors.

Ex:
- Earth
  - Americas
    - Canada
    - USA

If I filter on USA, `TreeNodeMultipleChoiceFilter` would only return Americas because USA (and its descendants - in this case none) is only a child of Americas. We would want that filter to return Americas and Earth because USA is a descendant of both. 

I find filtering any kind of tree descendants/ancestors on self-referential fields is confusing. How do you feel about changing all of the self-referential field filters to `NaturalKeyOrPKMultipleChoiceFilter` and wait for a use case for filtering those fields on ancestors/descendants?

Reply:
I'm fine with not implementing `ancestors` or `descendants` filters at this time.